mtags/src/main/scala-2.12/scala/meta/internal/pc/Compat.scala

```diff
@@ -16,4 +18,33 @@ trait Compat { this: MetalsGlobal =>
   def isAliasCompletion(m: Member): Boolean = false
 
   def constantType(c: ConstantType): ConstantType = c
+
+  def runOutline(files: OutlineFiles): Unit = {
+    this.settings.Youtline.value = true
+    runOutline(files.files)
+    if (files.firstCompileSubstitute) {
+      // if first compilation substitute we compile all files twice
```

Comment:
Does this work in case when we have for example 3 files:
A -> B -> C
that depend on each other? Could we add tests for that? We should be able to get member completions in A, which are influenced by type in B and C

Reply:
What if we use `this.addCompilationUnit` on all the files and then run typecheck. It would probably work if we compile all the files at the same time as the normal compiler

Reply:
Doesn't matter we only have
```scala
private[scala] def typeCheck(unit: RichCompilationUnit)
```
which type checks a single unit.